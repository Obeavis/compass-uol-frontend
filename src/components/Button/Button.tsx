import React, { FC } from "react";
import styled from "@emotion/styled";

const StyledButton = styled.button`
  outline: none !important;
  border: none;
  &:active {
    transform: translateY(2px);
  }
`;

export interface Props
  extends React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > {
  className?: string;
}

export const Button: FC<Props> = ({ className, ...rest }) => {
  return (
    <div className="w-full">
      <StyledButton className={`w-full text-gray-200 py-3 text-sm ${className}`} {...rest} />
    </div>
  );
};

export default Button;
