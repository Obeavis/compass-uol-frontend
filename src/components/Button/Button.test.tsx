import { render } from "@testing-library/react";
import Button from "./Button";

describe("Button component", () => {
  test("renders the component", async () => {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot();
  });
});
