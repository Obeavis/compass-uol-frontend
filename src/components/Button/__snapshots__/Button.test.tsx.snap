// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders the component 1`] = `
<div>
  <div
    class="w-full"
  >
    <button
      class="w-full text-gray-200 py-3 text-sm undefined css-1jlxohd"
    />
  </div>
</div>
`;
